hcl.html.markdown

```diff
@@ -0,0 +1,255 @@
+---
+category: tool
+tool: hcl
+contributors:
+    - ["Romans Malinovskis" , "http://github.com/romaninsh"]
+filename: terraform.txt
+---
+
+## Introduction
+
+HCL (Hashicorp Configuration Language) is a high-level configuration language used in tools from 
+Hashicorp (such as Terraform). This document focuses on a most recent HCL syntax (0.12).
+
+HCL is a declarative language and will consume all `*.tf` in the current folder. Sub-folders
+can be consumed through modules.
+
+This guide is focused on HCL specifics, you should be already familiar with what is Terraform.
+
+```hcl-terraform
+// Top-level HCL file will interactively ask user values for the variables
+// which do not have default value
+variable "ready" {
+  description = "Ready to learn?"
+  type = bool
+  // default = true
+}
+
+// Module block consults a specified folder for *.tf files, would
+// effectively prefix all resources id's with "module.learn-basics."
```

Comment:
it is just ids in this case, no apostrophe: http://www.sussex.ac.uk/informatics/punctuation/apostrophe/possessives